automatic/nexus-repository/tools/helpers.ps1

```diff
@@ -39,58 +38,122 @@ function Restore-NexusSSL {
     if (Test-Path "$BackupLocation\jetty-https.xml") {
       Copy-Item "$BackupLocation\jetty-https.xml" "$env:ProgramData\nexus\etc\jetty"
     }
+  }
+}
 
-    Write-Host "Nexus is now available with the restored SSL configuration"
+function Get-NexusCertificateDomain {
+  param(
+    [string]$DataDir = (Join-Path $env:ProgramData "sonatype-work"),
+
+    [string]$ProgramDir = (Join-Path $env:ProgramData "nexus")
+  )
+  $Config = Get-NexusConfiguration -Path $DataDir\nexus3\etc\nexus.properties
+  if ($Config.'nexus-args'.Split(',') -contains '${jetty.etc}/jetty-https.xml') {
+    [xml]$HttpsConfig = Get-Content $ProgramDir\etc\jetty\jetty-https.xml
+    $KeyToolPath = Join-Path $ProgramDir "jre/bin/keytool.exe"
+    $KeyStorePath = Join-Path (Join-Path $ProgramDir "etc/ssl") $HttpsConfig.SelectSingleNode("//Set[@name='KeyStorePath']").'#text'
+    $KeyStorePassword = $HttpsConfig.SelectSingleNode("//Set[@name='KeyStorePassword']").'#text'
+
+    if ((Test-Path $KeyToolPath) -and (Test-Path $KeyStorePath)) {
+      # Running in a job, as otherwise KeyTool fails when run without input
+      Start-Job {
+        $KeyToolOutput = $using:KeyStorePassword | & "$using:KeyToolPath" -list -v -keystore "$using:KeyStorePath" -J"-Duser.language=en" 2>$null
+        if ($KeyToolOutput -join "`n" -match "(?smi)Certificate\[1\]:\nOwner: CN=(?<Domain>.+?)\n") {
+          $Matches.Domain
+        }
+      } | Receive-Job -Wait
+    }
   }
 }
 
-function Wait-NexusAvailability {
+function Get-NexusUri {
```

Comment:
I had issues with this function. The cert on my Nexus server is a wild card, so it doesn't have a common name or anything on it that is _specific_ to the server nor that would match the given FQDN that I pass in as a package parameter.

Also, despite my given said FQDN, it ended up using the hostname. My _expectation_ as a user is if I provide an FQDN, then that should be used (regardless of what the cert may say... if I've provided it and I got it wrong, then it's my own fault for shooting myself in the foot (imho)).